lib/transbank/sdk/oneclick/mall/mall_transaction.rb

```diff
@@ -0,0 +1,97 @@
+module Transbank
+  module Webpay
+    module Oneclick
+      class MallTransaction
+        extend Transbank::Utils::NetHelper
+
+        AUTHORIZE_TRANSACTION_ENDPOINT = 'rswebpaytransaction/api/oneclick/v1.0/transactions'.freeze
+        TRANSACTION_STATUS_ENDPOINT = 'rswebpaytransaction/api/oneclick/v1.0/transactions/:buy_order'.freeze
+        TRANSACTION_REFUND_ENDPOINT = 'rswebpaytransaction/api/oneclick/v1.0/transactions/:buy_order/refunds'.freeze
+
+        class << self
+
+          def authorize(username:, tbk_user:, parent_buy_order:, details:,
+                        options: nil)
+            api_key = options&.api_key || default_integration_params[:api_key]
+            commerce_code = options&.commerce_code || default_integration_params[:commerce_code]
+            integration_type = options&.integration_type || default_integration_params[:integration_type]
+            base_url = integration_type.nil? ? Oneclick::Base::integration_type[:TEST] : Oneclick::Base.integration_type_url(integration_type)
+
+            url = base_url + AUTHORIZE_TRANSACTION_ENDPOINT
+            headers = webpay_headers(commerce_code: commerce_code, api_key: api_key)
+
+            details_array = details.map do |det|
+              hash = det.reduce({}) do |acc, (k, v)|
+                acc[k.to_s] = v
+                acc
+              end
+              # We need to make sure the hash's keys are strings (or symbols)
+              # since Ruby doesnt have hahes with indifferent access
+              # I chose to make them strings
+              details_hash = {}
+              details_hash[:amount] = hash.fetch('amount')
```

Comment:
Are you using `fetch` in order to throw a `KeyError` if the key isn't found?

Reply:
Yes, the logic being that we want to make the user know that they are missing a key before they call Transbank's API, where the missing key will make the operation fail, but not necessarily return a good error message (which is a problem we've had several times)